lib/resources/file.rb

```
@@ -129,31 +142,63 @@ def check_file_permission_by_mask(usergroup, flag)
       (file.mode & mask) != 0
     end
 
-    def check_file_permission_by_user(user, flag)
+    def usergroup_for(usergroup, specific_user)
+      if usergroup == 'others'
+        'other'
+      elsif (usergroup.nil? || usergroup.empty?) && specific_user.nil?
+        'all'
+      else
+        usergroup
+      end
+    end
+  end
+
+  class FilePermissions
+    attr_reader :inspec
+    def initialize(inspec)
+      @inspec = inspec
+    end
+  end
+
+  class UnixFilePermissions < FilePermissions
+    def check_file_permission_by_user(user, access_type, path)
+      flag = case access_type
+             when 'read'
+               'r'
+             when 'write'
+               'w'
+             when 'execute'
+               'x'
+             end
```

we should return an error if the access_type is unknown